### PR TITLE
tests Login GSMArena

### PR DESCRIPTION
**Pull request by amyrchyk**

### CODE REVIEW

File: src/main/java/com/qaprosoft/carina/demo/gui/components/LoginForm.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
+package com.qaprosoft.carina.demo.gui.components;
+
+import org.openqa.selenium.SearchContext;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.support.FindBy;
+
+import com.qaprosoft.carina.core.foundation.webdriver.decorator.ExtendedWebElement;
+import com.qaprosoft.carina.core.gui.AbstractUIObject;
+
+public class LoginForm extends AbstractUIObject{
```
Comment (vrudyk) [inline src/main/java/com/qaprosoft/carina/demo/gui/components/LoginForm.java:10]: add space before {

File: src/main/java/com/qaprosoft/carina/demo/gui/components/services/LoginService.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    public HomePage loginWithWorngEmail(String pass) {
+	String emailIncorect = RandomStringUtils.randomAlphanumeric(10) + "@gmail.com";
+	String errorMessage = "Reason: User record not found.";
+	HomePage homePage = new HomePage(getDriver());
+	homePage.open();
+	Assert.assertTrue(homePage.isPageOpened(), "Home page is not opened");
+	Header header = homePage.getHeeader();
+	LoginForm loginForm = header.openLoginForm();
+	loginForm.login(emailIncorect, pass);
+	Assert.assertEquals(loginForm.errorMassage(), errorMessage);
```
Comment (vrudyk) [inline src/main/java/com/qaprosoft/carina/demo/gui/components/services/LoginService.java:33]: add message

File: src/main/java/com/qaprosoft/carina/demo/gui/components/services/LoginService.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    public HomePage loginWithWorngPass(String email) {
+	String passIncorect = RandomStringUtils.randomAlphanumeric(10);
+	String errorMessage = "Reason: Wrong password.";
+	HomePage homePage = new HomePage(getDriver());
+	homePage.open();
+	Assert.assertTrue(homePage.isPageOpened(), "Home page is not opened");
+	Header header = homePage.getHeeader();
+	LoginForm loginForm = header.openLoginForm();
+	loginForm.login(email, passIncorect);
+	Assert.assertEquals(loginForm.errorMassage(), errorMessage);
```
Comment (vrudyk) [inline src/main/java/com/qaprosoft/carina/demo/gui/components/services/LoginService.java:46]: add message

File: src/main/java/com/qaprosoft/carina/demo/gui/components/LoginForm.java (hunk 1 above)
Comment (amyrchyk) [reply src/main/java/com/qaprosoft/carina/demo/gui/components/LoginForm.java]: done

File: src/main/java/com/qaprosoft/carina/demo/gui/components/services/LoginService.java (hunk 2 above)
Comment (amyrchyk) [reply src/main/java/com/qaprosoft/carina/demo/gui/components/services/LoginService.java]: done

File: src/main/java/com/qaprosoft/carina/demo/gui/components/services/LoginService.java (hunk 3 above)
Comment (amyrchyk) [reply src/main/java/com/qaprosoft/carina/demo/gui/components/services/LoginService.java]: done

File: src/main/java/com/qaprosoft/carina/demo/gui/components/LoginForm.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    public ExtendedWebElement passField;
+    
+    @FindBy(id="nick-submit")
+    public ExtendedWebElement logInButton;
+    
+    public LoginForm(WebDriver driver, SearchContext searchContext) {
+	super(driver, searchContext);
+    }
+
+    public void login(String email, String pass) {
```
Comment (vrudyk) [inline src/main/java/com/qaprosoft/carina/demo/gui/components/LoginForm.java:26]: better if this method will return LoginPage

File: src/main/java/com/qaprosoft/carina/demo/gui/components/services/LoginService.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    public HomePage loginWithWorngEmail(String pass) {
+	String emailIncorect = RandomStringUtils.randomAlphanumeric(10) + "@gmail.com";
+	String errorMessage = "Reason: User record not found.";
+	HomePage homePage = new HomePage(getDriver());
+	homePage.open();
+	Assert.assertTrue(homePage.isPageOpened(), "Home page is not opened");
+	Header header = homePage.getHeeader();
+	LoginForm loginForm = header.openLoginForm();
+	loginForm.login(emailIncorect, pass);
+	LoginPage loginPage = new LoginPage(getDriver());
```
Comment (vrudyk) [inline src/main/java/com/qaprosoft/carina/demo/gui/components/services/LoginService.java:35]: see above

File: src/main/java/com/qaprosoft/carina/demo/gui/components/services/LoginService.java [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    public HomePage loginWithWorngPass(String email) {
+	String passIncorect = RandomStringUtils.randomAlphanumeric(10);
+	String errorMessage = "Reason: Wrong password.";
+	HomePage homePage = new HomePage(getDriver());
+	homePage.open();
+	Assert.assertTrue(homePage.isPageOpened(), "Home page is not opened");
+	Header header = homePage.getHeeader();
+	LoginForm loginForm = header.openLoginForm();
+	loginForm.login(email, passIncorect);
+	LoginPage loginPage = new LoginPage(getDriver());
```
Comment (vrudyk) [inline src/main/java/com/qaprosoft/carina/demo/gui/components/services/LoginService.java:49]: see above

File: src/main/java/com/qaprosoft/carina/demo/gui/components/services/LoginService.java (hunk 5 above)
Comment (amyrchyk) [reply src/main/java/com/qaprosoft/carina/demo/gui/components/services/LoginService.java]: done

File: src/main/java/com/qaprosoft/carina/demo/gui/components/services/LoginService.java (hunk 6 above)
Comment (amyrchyk) [reply src/main/java/com/qaprosoft/carina/demo/gui/components/services/LoginService.java]: done

File: src/main/java/com/qaprosoft/carina/demo/gui/components/LoginForm.java (hunk 4 above)
Comment (amyrchyk) [reply src/main/java/com/qaprosoft/carina/demo/gui/components/LoginForm.java]: done

**Comment by vrudyk** [comment]
ok

**Pull request merged**